xprintidle.c

```diff
@@ -157,32 +156,34 @@ unsigned long workaroundCreepyXServer(Display *dpy, unsigned long idleTime) {
   CARD16 state;
   BOOL onoff;
 
-  if (DPMSQueryExtension(dpy, &dummy, &dummy)) {
-    if (DPMSCapable(dpy)) {
-      DPMSGetTimeouts(dpy, &standby, &suspend, &off);
-      DPMSInfo(dpy, &state, &onoff);
-
-      if (onoff) {
-        switch (state) {
-        case DPMSModeStandby:
-          /* this check is a little bit paranoid, but be sure */
-          if (idleTime < (unsigned)(standby * 1000))
-            idleTime += (standby * 1000);
-          break;
-        case DPMSModeSuspend:
-          if (idleTime < (unsigned)((suspend + standby) * 1000))
-            idleTime += ((suspend + standby) * 1000);
-          break;
-        case DPMSModeOff:
-          if (idleTime < (unsigned)((off + suspend + standby) * 1000))
-            idleTime += ((off + suspend + standby) * 1000);
-          break;
-        case DPMSModeOn:
-        default:
-          break;
-        }
-      }
-    }
+  if (!DPMSQueryExtension(dpy, &dummy, &dummy))
+    return idleTime;
+  else if (!DPMSCapable(dpy))
```

Comment:
Personally I prefer to have a plain "if" after a "returning if" instead of an "else if". But don't bother, this is also fine for me.